homeassistant/components/hassio/__init__.py

```diff
@@ -537,6 +537,8 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:  # noqa:
     if (data := await store.async_load()) is None:
         data = {}
 
+    assert isinstance(data, dict)
```

Comment:
Side note: Maybe we could make the `Store` class generic to be able to type the return value of `Store.async_load` and `Store.async_save`, similar to how we can type the update coordinator.

Reply:
I had a quick look - and that's going to be a big job!
See #74617